src/tests/ftest/io/nvme_io.py

```diff
@@ -76,7 +92,13 @@ def tearDown(self):
                 self.pool.disconnect()
                 self.pool.destroy(1)
         finally:
+            if self.agent_sessions:
+                AgentUtils.stop_agent(self.hostlist_clients,
```

Comment:
I'd recommend another try/finally here.  You still want to attempt to kill the daos servers and cleaning up the NVMe devices even if something goes wrong killing the agents